packages/core/src/exchanges/cache.ts

```diff
@@ -154,14 +158,18 @@ const afterQuery = (
   operationCache: OperationCache
 ) => (response: OperationResult) => {
   const { operation, data, error } = response;
+  const { additionalTypenames } = operation.context;
 
   if (data === undefined || data === null) {
     return;
   }
 
   resultCache.set(operation.key, { operation, data, error });
 
-  collectTypesFromResponse(response.data).forEach(typeName => {
+  [
```

Comment:
Nit: Could these go into a `generateTypenames` func? 

``` js
const generateTypenames = (responseData, additionalTypenames) => ...
```

Reply:
My concern with it is the unneeded size overhead this will introduce

Reply:
I need to become more size-concious 🙂